### ISSUE TITLE
Save Route not working in Internet Explorer

### ISSUE THREAD

**Issue by zstadler** [IsraelHikingMap/Site#138]
To reproduce:

- Open http://israelhiking.osm.org.il/#/13/31.7730/35.1200?s=w5udsmoyqL
- Click on the save button
- Click on any of the output formats

Rather than downloading the file, nothing happens.

First reported in [this Groopy thread](http://groopy.co.il/forummessage.aspx?messageid=405210)

**Comment by HarelM** [comment]
Seems to be a lot harder than I thought it will be...
I managed to make IE download a file but when try to open that file it fails...
Seems IE has issues with xml serializer which is implemented different than other browsers.
I'll see if I can work around that using the server...

**Comment by zstadler** [comment]
It works indeed!

However, when using the above example (http://israelhiking.osm.org.il/#/13/31.7730/35.1200?s=w5udsmoyqL) it seems like the new output is different.

Previously, the output was a track where each track point had lat, lon, and elevation:
```xml
  <trk>
    <name>Route 1</name>
    <desc>name=Route 1</desc>
    <trkseg>
      <trkpt lat="31.781882156411022" lon="35.07104930584319">
        <ele>552</ele>
      </trkpt>
      ...
```
and now the output is a route where each route point has lat, lon, and a synthetic name:
```xml
  <rte>
    <name>Route 1</name>
    <rtept lat="31.781882156" lon="35.071049306">
      <name>RPT001</name>
    </rtept>
      ...
```

According to the [GPX schema](http://www.topografix.com/GPX/1/1/), both `trkpt` and `rtept` are `wptType`. Therefore, no artificial names would be created. The point name will be taken from the user-given mane, if exists.

I also think the GPX should continue to produce a track rather than a route. If we want to extra picky, we can use the route format if routing was not performed by the site.

**Comment by HarelM** [comment]
I'll take a look, the point names are coming from gbsbabel but I'll make sure, if so I trust they know what they are doing...
The elevation is alarming...

**Comment by HarelM** [comment]
@zstadler I think I managed to fix most of the issues.
Regarding your concern about track and route.
Unfortunately GPSBabel does not support geojson so I needed a way to convert from geoJson and to geoJson so I written a small geoJson <==> GPX converter.
I implemented a very simple logic that does the following:
geoJson Point = GPX wpt (waypoint)
geoJson LineString = GPX rte (route)
geoJson MultiLineString = GPX trk (track)
Other types are currently not supported...
You can see the code here:
https://github.com/IsraelHikingMap/Site/blob/master/IsraelHiking.API/Gpx/GpxGeoJsonConverter.cs
If you think this is incorrect or partial I'd love to know as I couldn't find a good place to read about it...

**Comment by zstadler** [comment]
The difference between LineString and MultiLineString seems to be
structural. What's the semantic difference, if any?

Let's have a phone call about the semantics of internal objects and their
representations.

**Comment by HarelM** [comment]
Elevation fixed, removed unwanted names as part of 7f8936bf2f72c677589323707e00edcef050c96f and issue #140
